### PR TITLE
useReducedMotion: Accessibility hook to support prefers-reduced-motion

### PR DESCRIPTION
**Pull request by christianvuerings**
Adds `useReducedMotion` an accessibility hook to support prefers-reduced-motion

useReducedMotion

## Test Plan

Enable / disable the reduce motion settings on your OS and ensure the animation stops / starts.

### CODE REVIEW

File: docs/src/components/sidebarIndex.js [hunk 1]
```diff
@@ -31,6 +31,11 @@ const sidebarIndex: sidebarIndexType = [
     sectionPathname: 'setup',
     pages: ['GestaltProvider'],
   },
+  {
+    sectionName: 'Accessibility',
```
Comment (chrislloyd) [inline docs/src/components/sidebarIndex.js:35]: Should this be "Hooks" or similar? We don't have too many a11y components.

Comment (christianvuerings) [reply docs/src/components/sidebarIndex.js:35]: @chrislloyd Just wait for it, this is only the beginning... Also plan on exporting `useFocusVisible` and adding `VisuallyHidden` + `useId` soon

**Comment by AlbertCarreras** [comment]
@cvuerings Box does not support animation, so this hook is to be used with div tags or dangerouslyInlineStyle?

**Comment by christianvuerings** [comment]
> @cvuerings Box does not support animation, so this hook is to be used with div tags or dangerouslyInlineStyle?

It is to be used within Gestalt (if we add any animation) + in other repos where animation is used. That might be through div tags with `style`, `dangerouslyInlineStyle` on `Box`, using `className` or any other library they use.

File: packages/gestalt/src/useReducedMotion.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      setMatch(mediaQuery.matches);
+    };
+    handleChange();
+    mediaQuery.addEventListener('change', handleChange);
+    return () => {
+      mediaQuery.removeEventListener('change', handleChange);
+    };
+  }, [supportsMatchMedia]);
+  return matches;
+}
```
Comment (jennyscript) [inline packages/gestalt/src/useReducedMotion.js:27]: Feels like we should have a test for this though I guess it'd just be testing the browser API so maybe not

File: docs/src/useReducedMotion.doc.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const cards = [];
+const card = c => cards.push(c);
+
+card(
+  <PageHeader
+    name="useReducedMotion"
+    description={`
+      \`useReducedMotion\` uses the <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion" target="_blank">prefers-reduced-motion</a> CSS media feature to detect
+      if the user has requested that the system minimize the amount of non-essential motion.
```
Comment (jennyscript) [inline docs/src/useReducedMotion.doc.js:14]: Can we provide more guidance here on when devs should use this? Not everyone will click the link for a full explanation

Comment (christianvuerings) [reply docs/src/useReducedMotion.doc.js]: @jennyscript added - mind taking another look?

File: packages/gestalt/src/useReducedMotion.js (hunk 2 above)
Comment (christianvuerings) [reply packages/gestalt/src/useReducedMotion.js:27]: @jennyscript good callout - added tests.

@jennyscript @AlbertCarreras fyi this PR now also ads `@testing-library/react-hooks` to start testing hooks we export.

**Pull request merged**